src/model_scoring_functions.R

```diff
@@ -0,0 +1,219 @@
+  clades <- sort(unique(df_model_output$clade))
+  locs_modeled <- sort(unique(df_model_output$location))
+  
+  # Load validation data
+  df_validation <- arrow::read_parquet(paste0(hub_path, 
+                                              "target-data/oracle-output/nowcast_date=", 
+                                              ref_date, 
+                                              "/oracle.parquet"))
+  
+  # Pick out dates to score that were not used for any training
+  df_unscored <- read_csv(paste0(hub_path,
+                                 "/auxiliary-data/unscored-location-dates/", 
+                                 ref_date, 
+                                 ".csv"), 
+                          show_col_types = FALSE)
+  
+  # Load location data to match abbreviations to full name locations
+  load(paste0(hub_path, "auxiliary-data/hub_locations.rda"))
+  locs_join <- hub_locations |>
+    dplyr::select(abbreviation, location_name) |>
+    rename(location = location_name)
+  
+  df_unscored <- df_unscored |>
+    left_join(locs_join, by = c("location" = "location")) |>
+    select(abbreviation, target_date, count) |>
+    rename(location = abbreviation)
+  
+  # Look at subset where count == 0
+  df_unscored_zeros <- subset(df_unscored , count == 0)
+  
+  targets <- df_validation 
+  
+  # Max forecast date
+  forecast_date <- ref_date + 10 
+  
+  # Keep all forecast data
+  targets_forecast_temp <- targets |> 
+    subset(target_date > ref_date & target_date <= forecast_date)
+  
+  # Nowcast temp data frame to be refined
+  targets_nowcast_temp <- targets |>
+    subset(target_date <= ref_date)
+  
+  # Join by target_date and location - keeping only those 
+  # Merging "unscored" location data with full target data 
+  # Need only be done for dates <= ref_date
+  targets_nowcast_temp <- targets_nowcast_temp |>
+    right_join(y = df_unscored_zeros, by = join_by(target_date, location)) |>
+    select(-count)
+  
+  # Combine refined nowcast data with forecast data
+  targets <- rbind(targets_nowcast_temp, targets_forecast_temp) |>
+    subset(location %in% locs_modeled) |>
+    arrange(location, target_date)
+  
+  return(list(targets, df_model_output))
+}
+
+#' Support function for calculating energy scores given target and model_output data
+#' 
+#' @param targets data frame, of target data
+#' @param df_model_output data frame, of model output 
+#' 
+#' @return Returns a data frame containing energy scores by location and date
+calc_energy_scores <- function(targets, df_model_output){
+  # Energy Scores
+  columns <- c("es_score", "location", "target_date")
+  df_scores <- data.frame(matrix(nrow = 0, ncol = length(columns)))
+  colnames(df_scores) <- columns
+  
+  locs <- sort(unique(targets$location))
+  dates <- sort(unique(targets$target_date))
+  
+  # Loop over each day at each location
+  for(loc in locs){
+    for(day in dates){
+      
+      # For debugging and satisfaction
+      if(display = TRUE){
```

Comment:
This causes the script to fail

```suggestion
      if(display == TRUE){
```

Reply:
actually, this whole clause should go because `display` is not a parameter

Reply:
Indeed, it can go. A left over relic.